src/os.c

```diff
@@ -291,7 +291,16 @@ static void* mi_unix_mmap(size_t size, size_t try_alignment, int protect_flags)
       // TODO: if always failing due to permissions or no huge pages, try to avoid repeatedly trying?
       // Should we check this in _mi_os_init? (as on Windows)
       p = mi_unix_mmapx(size, try_alignment, protect_flags, lflags, fd);
+      #ifdef MADV_HUGEPAGE
+      if (p == MAP_FAILED) {
+        lflags &= MAP_HUGETLB; // large page could be set to madvise only
```

Comment:
Should that be `lflags &= ~MAP_HUGETLB` instead?

Reply:
I simplified it by getting back the original flag but indeed typo here :-)